xonsh/tools.py

```diff
@@ -428,16 +428,59 @@ def print_exception(msg=None):
     if 'XONSH_SHOW_TRACEBACK' not in env:
         sys.stderr.write('xonsh: For full traceback set: '
                          '$XONSH_SHOW_TRACEBACK = True\n')
```

Comment:
Interestingly, it _used_ to show up.  The idea was: if someone had explicitly set that value, we wouldn't show the message; but if it was just on the default `False`, we would show it.  I think we should keep that logic.  It's a little bit ugly, but we can check `if "XONSH_SHOW_TRACEBACK" no in env._d`.  Or we could be a little cleaner and add a new method to `xonsh.environ.Env` that does that check (by looking in `_d`) so at least we don't have to access the private attribute from outside.  Maybe `xonsh.environ.Env.is_manually_set(varname)`?